postgres/datadog_checks/postgres/postgres.py

```diff
@@ -387,20 +385,26 @@ def _connect(self):
         else:
             self.db = self._new_connection(self._config.dbname)
 
-    # Reload the track_activity_query_size setting on a new connection to the main db
-    def _load_query_max_text_size(self, db):
+    # Reload pg_settings on a new connection to the main db
+    def _load_pg_settings(self, db):
         try:
             with db.cursor(cursor_factory=psycopg2.extras.DictCursor) as cursor:
-                self.log.debug("Running query [%s]", TRACK_ACTIVITY_QUERY_SIZE_QUERY)
-                cursor.execute(TRACK_ACTIVITY_QUERY_SIZE_QUERY)
-                row = cursor.fetchone()
-                self._db_configured_track_activity_query_size = int(row['setting'])
-        except (psycopg2.DatabaseError, psycopg2.OperationalError) as e:
-            self.log.warning("cannot read track_activity_query_size from pg_settings: %s", repr(e))
+                self.log.debug("Running query [%s]", PG_SETTINGS_QUERY)
+                cursor.execute(
+                    PG_SETTINGS_QUERY,
+                    ("pg_stat_statements.max", "track_activity_query_size"),
+                )
+                rows = cursor.fetchall()
+                self.pg_settings.clear()
+                for setting in rows:
+                    name, val = setting
+                    self.pg_settings[name] = val
+        except (psycopg2.DatabaseError, psycopg2.OperationalError) as err:
+            self.log.warning("Failed to query for pg_settings: %s", repr(err))
             self.count(
                 "dd.postgres.error",
                 1,
-                tags=self._config.tags + ["error:load-track-activity-query-size"] + self._get_debug_tags(),
+                tags=self._config.tags + ["error:load-pg-settings"] + self._get_debug_tags(),
```

Comment:
Do we have any users with monitors configured to the previous tag value? We should double check because this would change effect monitors

Reply:
Nice catch, @ChristineTChen! Just double checked and **no** orgs are using this, this is also an internal metric so users shouldn't expect them to be stable, nonetheless, good call!